src/MoBi.Core/Commands/SetParentPathCommand.cs

```diff
@@ -30,9 +32,46 @@ protected override void ExecuteWith(IMoBiContext context)
       {
          base.ExecuteWith(context);
          _container.ParentPath = _newParentPath;
+         updateObjectPathsInSpatialStructure();
          Description = AppConstants.Commands.UpdateParentPath(_container.EntityPath(), (_newParentPath ?? new ObjectPath()).ToPathString());
       }
 
+      private void updateObjectPathsInSpatialStructure()
+      {
+         //should probably never be null
+         var spatialStructure = _buildingBlock as SpatialStructure;
+         if (spatialStructure == null)
+            return;
+
+         var oldObjectPathString = createContainerPath(_oldParentPath);
+         var newObjectPathString = createContainerPath(_newParentPath);
+
+         bool referencesPath(ObjectPath objectPath) => objectPath != null && objectPath.PathAsString.StartsWith(oldObjectPathString);
+         
+         //we know that the object path exists in this context
+         void updatePaths(ObjectPath objectPath) => objectPath.ReplaceWith(objectPath.PathAsString.Replace(oldObjectPathString, newObjectPathString).ToPathArray());
```

Comment:
not sure if there is a better way of doing this. 

How to find a path in the hierarchy. ? it starts with the oldPath

How to update the path? I have to replace the old path with the new one. 
The constant PathAsString is probably not the most efficient. But one can go from A|B to A or to A|B|C so you can't really use indexes etc..

Reply:
I would do rather something like this:
```
void updatePaths(ObjectPath objectPath)
{
   var objectPathAsString = objectPath.PathAsString;
   if (!objectPathAsString.StartsWith(oldParentContainerPathString)
      return;
   
   objectPathAsString = newParentContainerPathString + objectPathAsString.Substring(oldParentContainerPathString.Length);
  objectPath.ReplaceWith(objectPathAsString.ToPathArray());
}
```

This checks, that the object paths **starts** with the old parent container path and replaces **only the first occurrence** of the old parent container path with the new one.
So if the old container path was A and the new container path is B:
_**A**|X|A|Y_ will be renamed to _**B**|X|A|Y_
_C|X|A|Y_ will be not modified

Reply:
This referencePath already checks this. We know it starts with the path

Reply:
Replace also only replaces the first occurrence I think but I can add a test to prove this. I don't we need to change the code

Reply:
> Replace also only replaces the first occurrence I think

it replaces all occurrences. It's very unlikely situation that we would have a situation like this, but you know... "should never happen" 😉 

```
void updatePath(ObjectPath objectPath) => 
 objectPath.ReplaceWith(newContainerPathString + objectPath.PathAsString.Substring(oldContainerPathString.Length)).ToPathArray());
```